libraries/zookeeper_service.rb

```diff
@@ -86,6 +86,7 @@ class ZookeeperService < Chef::Provider
       include PoiseService::ServiceMixin
 
       def action_enable
+        new_resource.notifies(:restart, new_resource, :delayed)
```

Comment:
I spoke with @coderanger here about this. The gist of it is that any of the resources that change in the notifying block should kick the service. The most apparent would be the upgrade of a package.